flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/SnapshotType.java

```diff
@@ -25,6 +25,8 @@ public interface SnapshotType extends Serializable {
 
     boolean isSavepoint();
 
+    boolean isIncrementalCheckpoint();
```

Comment:
It's good to name this `isFull` or `isIncremental` for internal APIs. What I meant in the email is that the REST API should has `incremental` or `is_incremental` for input or output, which is user-facing. To simplify, I suggest name this `isFull()`.

And the default implementation would be `getSharingFilesStrategy() != FORWARD_BACKWARD` for `isFull()`. It is good for maintain as you won't change this part when adding another instance other than `FULL_CHECKPOINT`. WDYT?

Reply:
>It's good to name this isFull or isIncremental for internal APIs. What I meant in the email is that the REST API should has incremental or is_incremental for input or output, which is user-facing. To simplify, I suggest name this isFull().

👍 

>And the default implementation would be getSharingFilesStrategy() != FORWARD_BACKWARD for isFull(). It is good for maintain as you won't change this part when adding another instance other than FULL_CHECKPOINT. WDYT?

👍